pkg/registry/memory/storage.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memory
+
+import "github.com/networkservicemesh/sdk/pkg/tools/syncmap"
+
+// Storage provides memory access for NSM resources
+type Storage struct {
```

Comment:
Why create a separate struct here... why not just have the maps directly in the chain element?

Reply:
NetworkServeRegistry and ServiceDiscoveryServer depend on all NSM resources{ns, nsm, nse}. I used `Storage` to minimize of passed parameters.

Reply:
Right... but why are you passing them at all?

Reply:
NOTE: We can not create these maps internally because of these maps should be shared between different memory registry services such as `Discovery`, `NetworkSerivceManager`, `NetworkServiceEndpoints`.

Reply:
Note how this is handled in the [monitor chain element](https://github.com/networkservicemesh/sdk/blob/master/pkg/networkservice/common/monitor/server.go#L49)

Reply:
Discussion moved to comment above https://github.com/networkservicemesh/sdk/pull/241#discussion_r429558410